PlaceRouteHierFlow/placer/ILP_solver.cpp

```diff
@@ -2795,10 +2915,15 @@ double ILP_solver::GenerateValidSolution(const design& mydesign, const SeqPair&
         }
         if (non_zero_yoffset) break;
       }
-      if (!non_zero_yoffset) roundup(Blocks[i].y, y_pitch);
+      if (!non_zero_yoffset && Blocks[i].y % y_pitch != 0) {
+        snapGridILP = true;
+        break;
+      }
     }
   }
 
+  if (snapGridILP && !GenerateValidSolutionCore(mydesign, curr_sp, drcInfo, num_threads, true)) return -1;
```

Comment:
@854768750 This call here repeats ILP if any block is off-grid and snaps them back to grid. We don't want to go into placeongrid mode always [`here`](https://github.com/ALIGN-analoglayout/ALIGN-public/blob/80b4604ad42e713dca42715ce29d9d1565c8275a/PlaceRouteHierFlow/placer/design.cpp#L79-L84). That is costing too much in runtime.

Reply:
ok, my understanding is that:
1. If no block has offset, then no need to snapGridILP. In this case, rounding after compaction is required.
2. If any block has offset, snapGridILP is true. Then no need to do rounding

Reply:
In this testcase, no block has offset, then snapGridILP is false, compaction does not consider on grid, and rounding was removed, so some block are off grid.
Some options:
1. Always add offset==0 for empty offset, to activate on grid compaction
2. resume rounding for empty offset

Reply:
> ok, my understanding is that:
> 
>     1. If no block has offset, then no need to snapGridILP. In this case, rounding after compaction is required.
> 
>     2. If any block has offset, snapGridILP is true. Then no need to do rounding

I removed compaction totally.
If there is no offset constraint and if any block is off-grid, we call ILP a second time setting snap to grid in ILP mode.
If any offset is present for any block, we just ever to the zero offset constraint that you had added in the previous PR.